Add --query example for describe-instance-patches

jacklin213 opened the pull request:
This commit adds an example usage of using --query to retrieve information based on "InstalledTime" as it is not a valid filter key.

Can also be used for the "Title" of the patch as that is also not a valid filter key

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

justindho commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
     "NextToken": "--token string truncated--"
   }
+
+**To get a list of patches installed patches since a specified InstalledTime for an instance**
```
Looks like an extra word was accidentally added here.
```suggestion
**To get a list of patches installed since a specified InstalledTime for an instance**
```

jacklin213 replied:
Removed extra word in latest rebase

justindho commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
+
+**To get a list of patches installed patches since a specified InstalledTime for an instance**
+  
+This example combines the use of --filters and --query to retrieve information about patches installed since a specified time for the specified instance.
```
It would be nice if we could follow similar wording to the other [examples](https://docs.aws.amazon.com/cli/latest/reference/ssm/describe-instance-patches.html#examples) in this file. The suggestion also appends `::` to follow rst formatting.

```suggestion
The following ``--describe-instance-patches`` example retrieves information about patches... ::
```

jacklin213 replied:
Updated formatting to match the wording styles from other examples

justindho commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
+  
+This example combines the use of --filters and --query to retrieve information about patches installed since a specified time for the specified instance.
+
+Note: The format of InstalledTime is in epoch time. The human readable format of 1546992000.0 is 2019-01-09 12:00:00 UTC.
```
This is our fault for taking so long to review this, but do you mind updating the date to something more recent?

justindho replied:
This is our fault for taking so long to review this, but do you mind updating the date to something more recent?

justindho commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
+
+Note: The format of InstalledTime is in epoch time. The human readable format of 1546992000.0 is 2019-01-09 12:00:00 UTC.
+
+Command::
```
Following the same format as existing examples, this line can now be removed.

jacklin213 replied:
Updated in latest rebase

justindho commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
+
+Note: The format of InstalledTime is in epoch time. The human readable format of 1546992000.0 is 2019-01-09 12:00:00 UTC.
+
+Command::
```
Following the same format as existing examples, this line can now be removed.

jacklin213 replied:
Updated in latest rebase

justindho commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
+
+Command::
+
+  aws ssm describe-instance-patches --instance-id "i-1234567890abcdef0" --filters Key=State,Values=Installed --query "Patches[?InstalledTime >= `1546992000.0`]"
```
The formatting of commands has been updated since this PR was first opened. Similar to the other examples in this file, could we update to use `\` to break up the options into their own lines?

jacklin213 replied:
Updated in latest rebase

justindho commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
+
+Command::
+
+  aws ssm describe-instance-patches --instance-id "i-1234567890abcdef0" --filters Key=State,Values=Installed --query "Patches[?InstalledTime >= `1546992000.0`]"
```
The formatting of commands has been updated since this PR was first opened. Similar to the other examples in this file, could we update to use `\` to break up the options into their own lines?

jacklin213 replied:
Updated in latest rebase

justindho reviewed (commented):
@jacklin213 Thanks for the contribution. Apologies for the delay in getting back to you. I just had some small comments regarding wording and formatting, since we now have updated formatting standards for our documentation. The branch will also need a rebase off of the latest `develop` branch.

Would you still be interested in making these updates to get this PR merged?

jacklin213 commented:
Hey thanks for getting back to me on this.
Apologies for the delayed response, not sure why I didn't get a notification when you commented.

I will revisit this PR this week and see if it is still needed. If yes I will go ahead with the rebase and make updates as requested in the feedback

jacklin213 commented on awscli/examples/ssm/describe-instance-patches.rst:
```diff
@@ -55,3 +55,30 @@ Output::
+
+**To get a list of patches installed patches since a specified InstalledTime for an instance**
+  
+This example combines the use of --filters and --query to retrieve information about patches installed since a specified time for the specified instance.
```
Updated formatting to match the wording styles from other examples

jacklin213 commented:
Hey team,

Apologies for the delay on this one. I have rebased with latest commits in `develop` and taken in the improvements feedback provided by @justindho

While doing so also addressed a few other issues within the file:
- Modifying the output of `InstalledTime` to be in ISO8601 as By default, the AWS CLI version 2 translates all response DateTime values to ISO 8601 format. [Source](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-parameters-types.html#parameter-type-timestamp)
- Replaced `"InstalledTime": 0.0` in Example 2 with `"InstalledTime": "1970-01-01T00:00:00+00:00"` as that is how SSM handles `0`.
- Fixed some spacing issues for consistency. Example 1 and examples from other files use 4 spaces while example 2 had 2 spaces only

Let me know if any further adjustments need to be made.
Command was tested on `aws-cli/2.9.6`

elysahall reviewed (approved):
LGTM

jacklin213 commented:
Looks like we might need another approver for this one?

kyleknap reviewed (approved):
Thanks! Merging.
